### PR TITLE
Clarify bean ordering when iterating over an instance

### PR DESCRIPTION
**Pull request by rmanibus**
cc @mkouba

### CODE REVIEW

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 1]
```diff
@@ -957,6 +957,27 @@ Alternatively, you can use the `@LookupIfProperty` and `@LookupUnlessProperty` a
  }
 ----
 
+if there is more than one bean that matches the required type and qualifiers and is eligible for injection, it is possible to iterate (or stream) available bean instances.
```
Comment (gsmet) [inline docs/src/main/asciidoc/cdi-reference.adoc:960]:
```suggestion
If there is more than one bean that matches the required type and qualifiers and is eligible for injection, it is possible to iterate (or stream) available bean instances.
```

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 2]
```diff
@@ -957,6 +957,27 @@ Alternatively, you can use the `@LookupIfProperty` and `@LookupUnlessProperty` a
...
+
+    void printServiceName() {
+        if(service.isAmbiguous()){
+            for (Service service : serviceInstance) {
+
+            }
+        }
+    }
+ }
+ ----
```
Comment (gsmet) [inline docs/src/main/asciidoc/cdi-reference.adoc:979]: I'm not sure if it's just GitHub UI but it looks as if there was an additional space in front of these lines?

If so, might explain why the CI build failed.

**Review by gsmet — COMMENTED** [review]
Thanks! I added a few suggestions and will let @mkouba review.

File: docs/src/main/asciidoc/cdi-reference.adoc (hunk 1 above)
Comment (mkouba) [inline docs/src/main/asciidoc/cdi-reference.adoc:960]: I think that this paragraph would deserve a separate section with title along the lines of _"Sorting beans obtained with programmatic lookup"_.

Comment (manovotn) [reply docs/src/main/asciidoc/cdi-reference.adoc:960]: I agree, I'd add it as a separate non-standard feature.
Ideally with a use case (why you might need it) and with an example showing two prioritized beans (and third with no prio?) and what the ordering looks like.

Comment (rmanibus) [reply docs/src/main/asciidoc/cdi-reference.adoc]: @mkouba , just updated the PR, WDYT ?

**Review by mkouba — APPROVED** [review]

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 3]
```diff
@@ -957,6 +957,47 @@ Alternatively, you can use the `@LookupIfProperty` and `@LookupUnlessProperty` a
...
+
+If there is more than one bean that matches the required type and qualifiers and is eligible for injection, it is possible to iterate (or stream) available bean instances.
+Beans returned by both stream and iterator methods are sorted by priority as defined by `io.quarkus.arc.InjectableBean#getPriority()`. Higher priority goes first.
+If no priority is explicitly declared, 0 is assumed.
+
+[source,java]
+----
+@Priority(100)
+@ApplicationScoped
+class FirstService {
```
Comment (manovotn) [inline docs/src/main/asciidoc/cdi-reference.adoc:970]:
```suggestion
class FirstService implements Service {
```

Same goes for the other two impls.

Comment (mkouba) [reply docs/src/main/asciidoc/cdi-reference.adoc:970]: That's a good point but in that case we should also add `interface Service {}` to the snippet...

Comment (rmanibus) [reply docs/src/main/asciidoc/cdi-reference.adoc]: done !

**Review by manovotn — APPROVED** [review]

**Pull request merged**